### ISSUE TITLE
_NET_WM_STRUT_PARTIAL is not set in xmonad

### ISSUE THREAD

**Issue by yunaphung** [LemonBoy/bar#20]
Linux Mint Cinnamon 15
Xmonad version 0.11

xmonad won't acknowledge the existence of bar as a result of this.

If I'm not mistaken, http://hackage.haskell.org/packages/archive/xmonad-contrib/0.11/doc/html/src/XMonad-Hooks-EwmhDesktops.html shows `_NET_WM_STRUT` is supported, but not `_NET_WM_STRUT_PARTIAL`. Using xprop on the bar shows that only `_NET_WM_DESKTOP` is set.

I changed line 256 from

```
    cookies[3] = xcb_intern_atom (c, 0, strlen ("_NET_WM_STRUT_PARTIAL")   , "_NET_WM_STRUT_PARTIAL");
```

to

```
    cookies[3] = xcb_intern_atom (c, 0, strlen ("_NET_WM_STRUT")   , "_NET_WM_STRUT");
```

Running xprop on the bar shows that `_NET_WM_STRUT` gets set. However this doesn't fix it either. As seen [here](http://standards.freedesktop.org/wm-spec/wm-spec-1.3.html#id2732272), it states that `_NET_WM_STRUT_PARTIAL` needs to be set in order to reserve space. Looking at [XMonad.Hooks.ManageDocks](http://hackage.haskell.org/packages/archive/xmonad-contrib/0.10/doc/html/src/XMonad-Hooks-ManageDocks.html), there's a `getStrut` which does account for `_NET_WM_STRUT`, but my understanding of haskell is too limited to understand it.

 Also, xmonad just tiles the bar since the hacky thing on line 351 isn't run.

```
    xcb_change_window_attributes (c, win, XCB_CW_OVERRIDE_REDIRECT, (const uint32_t []){ !ewmh_docking });
```

**tl;dr:** I have no idea what I'm doing.
